dtfs-central-api/test-helpers/test-data/payment-matching-tolerances.ts

```diff
@@ -1,16 +1,9 @@
 import { CURRENCY, PaymentMatchingToleranceEntityMockBuilder } from '@ukef/dtfs2-common';
+import { getIncrementingPositiveIntegerIdGenerator } from '../get-integer-id-generator';
 
 export const aListOfZeroThresholdActivePaymentMatchingTolerances = () => {
-  const toleranceIdGenerator = idGenerator();
+  const toleranceIdGenerator = getIncrementingPositiveIntegerIdGenerator(1);
```

Comment:
What's the reason for introducing `getIncrementingPositiveIntegerIdGenerator` instead of using `idGenerator`?

Reply:
Because `idGenerator` is not very specific. When defined inside the test file, it is easy enough to figure out what it is doing, but having an export called `idGenerator` could mean anything. The use of `get` is to communicate that calling this function returns a generator to you

Reply:
Do we now have multiple ID generators used for testing purposes? Maybe could call it `getIdGenerator`, I don't think that's too generic.

IMO, `getIncrementingPositiveIntegerIdGenerator` is too long and descriptive.

Reply:
Before SQL, id in this project would usually refer to an `ObjectId`. It is because of SQL that id can now mean integer or `ObjectId`, hence the need to distinguish between them so that a future developer does not use this generator expecting it to produce `ObjectId`s

Reply:
The 2x function names do not highlight this at all, neither `idGenerator` or `getIncrementingPositiveIntegerIdGenerator` say to me "could be an integer or `ObjectId`", it just reads "i'm generating an incremental ID" really.

Should we maybe have 1x ID generator that conditionally creates a different ID?

Reply:
`getIncrementingPositiveIntegerIdGenerator` communicates to me that you create a generator which yields positive integers which increment after each `.next()` is called. 

`idGenerator` just communicates that it is an id generator (which it technically isn't because you have to call it to get the `Generator` object) which is lacking context around what it means by id

Reply:
Perhaps a compromise would be to go with `getSqlIdGenerator` and then that makes it clear and disambiguates it from other ids we have in use in the codebase? And if we ever wanted another incrementing positive integer id generator for a different kind of id, we could make a second helper and have both the sql one and the other one point to the same function which does that behind the scenes?

Reply:
I have tried to stay away from `SQL id` like names because an SQL id does not need to be an integer (for example, our `FacilityUtilisationData` table has a primary key which is `NVARCHAR(255)`). However, for most cases we will be using integers for SQL ids so I am happy with that name